addons/mail/static/src/models/user/user.js

```diff
@@ -35,6 +55,70 @@ function factory(dependencies) {
             return partner.nameOrDisplayName;
         }
 
+        /**
+         * Opens a chat between the partner of this user and the current partner
+         * and returns it.
+         *
+         * If a chat already exists for those partners it is reused. Otherwise a
+         * new chat is created.
+         *
+         * If a chat is not appropriate a notification is displayed instead.
+         *
+         * @param {Object} [params]
+         * @returns {mail.thread|undefined}
+         */
+        async openChat(params) {
+            if (!this.partner) {
+                await this.async(() => this.fetchPartner());
+            }
+            if (!this.partner) {
+                this.env.services['notification'].notify({
+                    message: this.env._t("You can only chat with existing users."),
+                    type: 'warning',
+                });
+                return;
+            }
+            // in other cases a chat would be valid, find it or try to create it
+            let chat = this.env.models['mail.thread'].find(thread =>
+                thread.model === 'mail.channel' &&
+                thread.channel_type === 'chat' &&
+                thread.correspondent === this.partner
+            );
+            if (!chat) {
+                chat = await this.async(() =>
+                    this.env.models['mail.thread'].createChat({
+                        partnerIds: [this.partner.id],
+                    })
+                );
+            }
+            if (!chat) {
+                this.env.services['notification'].notify({
+                    message: this.env._t("An unexpected error occurred during the creation of the chat."),
+                    type: 'warning',
+                });
+                return;
+            }
+            await chat.open(params);
+            return chat;
+        }
+
+        /**
+         * Opens the most appropriate view that is a profile for this user.
+         */
+        async openProfile() {
+            if (!this.partner) {
+                await this.async(() => this.fetchPartner());
+            }
+            if (!this.partner) {
```

Comment:
add comment this can only happen if user was inserted with a bad id, might be worth calling this.delete then